<a href="https://colab.research.google.com/github/yoonchowon/myfirstcode/blob/main/CTBC_9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. 두가지 이상의 선택, 다방향 선택구조 코딩
 - 배송료 처리 3가지로 코딩
 - 변수 : price
 - 조건 : 10만원 이상은 배송비 0.2만원 이상은 3천원,2만원 이하는 5천원

In [ ]:
price = int(input("구입금액 입력:"))           # 변수 price 사용자가 정수형 금액입력
if price >= 100000:                            # price의 값이 10만원보다 크거나 같으면
  shipplng_cost = 0                            # 선택1. 배송료(shipping_cost)는 0원
else:                                          # 그게 아니면
  if price >=20000:                            # 선택2.price값이 2만원보다 크거나 같으면
    shipping_cost = 3000                       # 배송료(shipping_cost)는 3000원
  else:                                        # 그것도 아니면
     shipping_cost = 5000                      # 선택3. 배송료(shipping_cost)는 5천원
print(f'배송료는 {shipping_cost}원 입니다.')   # F-String문자 Format으로 최종 배송료 출력



구입금액 입력:20000
배송료는 3000원 입니다.



1. "구입금액 입력:"이라는 메시지를 출력하고 사용자로부터 구입금액을 입력받습니다.
2. 입력받은 구입금액을 정수형으로 변환하여 price 변수에 저장합니다.
3. 조건문을 사용하여 배송비(shipping_cost)를 계산합니다.
- 만약 price가 100000 이상이라면, 배송비는 0원입니다. (shipping_cost = 0)
- 그렇지 않고 price가 100000 미만이고 20000 이상이라면, 배송비는 3000원입니다. (shipping_cost = 3000)
- 그렇지 않으면 (즉, price가 20000 미만인 경우), 배송비는 5000원입니다. (shipping_cost = 5000)
4. shipping_cost의 값을 출력하여 배송료를 알려줍니다. (f'배송료는 {shipping_cost}원 입니다.')

3.0부터 9까지 출력하는 코드 작성
 - 변수 : x

In [ ]:
x = 0                   # 변수 x를 0으로 초기화합니다.
while x < 10:           # x가 10보다 작은 동안 반복합니다.
  #print(x)# print(x)   # x의 값을 출력합니다. (주석 처리된 부분)
  print(x, end='')      # x의 값을 출력하고 줄바꿈을 하지 않고 출력합니다.
  x = x + 1             # x의 값을 1씩 증가시킵니다.

0123456789

위의 주석을 참고하시면 됩니다. 주석 처리된 부분은 print(x)로 x의 값을 출력하는 부분인데, 주석 처리되어 있어서 실제로는 실행되지 않습니다. print(x, end='')는 x의 값을 출력하고 줄바꿈을 하지 않고 출력하는 방식입니다. 그리고 x = x + 1은 x의 값을 1씩 증가시키는 부분입니다.

따라서 위의 코드는 0부터 9까지의 숫자를 한 줄에 출력하는 역할을 합니다. 출력 결과는 0123456789가 됩니다. 추가 설명이 필요하거나 다른 질문이 있으시면 언제든지 말씀해주세요!

4. 1부터 10까지 합산하는 코드 작성

In [ ]:
x = 1                  # 변수 x를 1로 초기화합니다.
total = 0              # 변수 total을 0으로 초기화합니다.
while x <= 10:         # x가 10 이하인 동안 반복합니다.
  total+= x            # total에 x 값을 더합니다. (total = total + x)
  #total = total + 1   # total에 1을 더합니다. (주석 처리된 부분)
  #x = x + 1           # x에 1을 더합니다. (주석 처리된 부분)
  x += 1               # x에 1을 더합니다.
print(total)           # total의 최종 값을 출력합니다.

55


위의 주석을 참고하시면 됩니다. 주석 처리된 부분은 total = total + 1과 x = x + 1로, 각각 total과 x를 1씩 증가시키는 부분입니다. 주석 처리되어 있어서 실제로는 실행되지 않습니다. total += x는 total에 x 값을 더하는 부분입니다. x += 1은 x에 1을 더하는 부분입니다.

따라서 위의 코드는 1부터 10까지의 숫자를 더한 결과를 출력합니다. 출력 결과는 55가 됩니다.

5. 3-6-9 게임 (3의 배수 판단): 3의 배수를 3으로 나누면 나머지가 0이 되므로 (i mod 3)의 값이 0이면, 박수를 출력,아니면 숫자 출력하는 코드


In [ ]:
for num in range(1, 11):   # 1부터 10까지의 숫자에 대해 반복합니다.
  if num % 3== 0:          # 현재 숫자가 3으로 나누어 떨어지는지 확인합니다.
    print("박수",end='')   # 3의 배수인 경우 "박수"를 출력하고 줄바꿈을 하지 않고 출력합니다.
  else:
      print(num, end='')   # 3의 배수가 아닌 경우 숫자를 출력하고 줄바꿈을 하지 않고 출력합니다.

12박수45박수78박수10

위의 주석을 참고하시면 됩니다. 주어진 코드는 1부터 10까지의 숫자에 대해 반복하면서 각 숫자가 3의 배수인지 확인하고, 3의 배수인 경우 "박수"를 출력하고 나머지 숫자는 그대로 출력하는 역할을 합니다. 출력 결과는 "1박수2박수4박수5박수7박수8박수10"입니다.

In [ ]:
num = 1                      # 변수 num을 1로 초기화합니다.
while num <= 20:             # num이 20 이하인 동안 반복합니다.
  if num % 3 == 0:           # 현재 num이 3의 배수인지 확인합니다.
    print("박수", end='')    # 3의 배수인 경우 "박수"를 출력하고 줄바꿈을 하지 않고 출력합니다.
  else:
      print(num, end='')     # 3의 배수가 아닌 경우 숫자를 출력하고 줄바꿈을 하지 않고 출력합니다.
  num += 1                   # num을 1씩 증가시킵니다.

12박수45박수78박수1011박수1314박수1617박수1920

위의 주석을 참고하시면 됩니다. 주어진 코드는 1부터 20까지의 숫자에 대해 반복하면서 각 숫자가 3의 배수인지 확인하고, 3의 배수인 경우 "박수"를 출력하고 나머지 숫자는 그대로 출력하는 역할을 합니다. 출력 결과는 "12박수45박수78박수101112박수1415박수171819박수"입니다.

In [ ]:
print("환영합니다.") # "환영합니다."를 출력합니다.
print("환영합니다.") # "환영합니다."를 출력합니다.
print("환영합니다.") # "환영합니다."를 출력합니다.
print("환영합니다.") # "환영합니다."를 출력합니다.
print("환영합니다.") # "환영합니다."를 출력합니다.

환영합니다.
환영합니다.
환영합니다.
환영합니다.
환영합니다.


주어진 코드는 print("환영합니다.")를 5번 반복하여 "환영합니다."를 출력하는 역할을 합니다. 따라서 위의 코드는 "환영합니다."를 5번 출력합니다.

In [ ]:
 print("환영합니다.\n" * 5) # "환영합니다."라는 문장을 5번 반복하여 출력하는 코드입니다

환영합니다.
환영합니다.
환영합니다.
환영합니다.
환영합니다.



주어진 코드는 "환영합니다.\n"을 5번 반복하여 출력합니다. \n은 줄바꿈을 나타내는 이스케이프 문자입니다. 따라서 위의 코드는 "환영합니다."를 5번 출력하고 각 문장 사이에 줄바꿈을 추가하여 출력합니다.

In [ ]:
count = 0                                   # 변수 count를 0으로 초기화합니다.
while count <=5:                            # count가 5 이하인 동안 반복합니다.
  print("(",count, ")", "환영합니다.")      # count 값을 출력하고 "환영합니다."를 출력합니다.
  count = count + 1                         # count를 1씩 증가시킵니다.

( 0 ) 환영합니다.
( 1 ) 환영합니다.
( 2 ) 환영합니다.
( 3 ) 환영합니다.
( 4 ) 환영합니다.
( 5 ) 환영합니다.


위의 주석을 참고하시면 됩니다. 주어진 코드는 count 값을 출력하고 "환영합니다."를 출력하는 역할을 합니다. count는 0부터 시작하여 1씩 증가하며, 반복문은 count가 5 이하일 때까지 실행됩니다. 따라서 위의 코드는 "(0) 환영합니다.", "(1) 환영합니다.", ..., "(5) 환영합니다."를 출력합니다.